store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -9999,7 +9999,11 @@ protected void endTransaction(boolean success) throws ServiceException {
                 if (deletes.blobs != null) {
                     // delete any blobs associated with items deleted from db/index
                     for (MailboxBlob blob : deletes.blobs) {
```

Comment:
in what scenario we are getting null blob from deletes.blobs list?

Reply:
Tag, Task, contact, Delete Contact Group

Reply:
https://jira.corp.synacor.com/browse/ZCS-11938?focusedCommentId=2022326&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-2022326

`Issues with:`
- Permanent delete appointment
- Permanent delete contact/contact group
- Delete tag

`Works fine:`
- Permanent delete mail
- Permanent delete briefcase file
- Save search

@prashantajari can add more if anything missing in above. Client automation caught only above.

Reply:
Thanks Jitesh, I have tried above test cases and now its workign fine with fix.

Reply:
@RakeshAMore It's good not reproducible any more, from the scenarios I meant to understand when is the blob missing to delete for any of the Tag, Task, contact, Delete Contact Group

Reply:
I think you take it wrong :D, I did not say this is not reproducible, I said it is working fine **with fix**. :)